File: _includes/code/howto/manage-data.classes.ts
// How-to: Manage-data -> Classes - TypeScript examples
// run with: node --loader=ts-node/esm FILENAME.ts
import assert from 'assert';

// ================================
// ===== INSTANTIATION-COMMON =====
// ================================
import weaviate from 'weaviate-ts-client';

const client = weaviate.client({
  scheme: 'http',
  host: 'anon-endpoint.weaviate.network',
  headers: {
    'X-OpenAI-Api-Key': process.env['OPENAI_API_KEY'],
  },
});

// START CreateCollection  // START ReadOneCollection  // START UpdateCollection
const className = 'Article';

// END CreateCollection  // END ReadOneCollection  // END UpdateCollection

// ================================
// ===== CREATE A CLASS =====
// ================================

// Clean slate
try {
  await client.schema.classDeleter().withClassName(className).do();
} catch (e) {
  // ignore error if class doesn't exist
}

// START CreateCollection
const emptyClassDefinition = {
  class: className,
};

// Add the class to the schema
let result = await client
  .schema
  .classCreator()
  .withClass(emptyClassDefinition)
  .do();

// The returned value is the full class definition, showing all defaults
console.log(JSON.stringify(result, null, 2));
// END CreateCollection

// Test
console.assert('invertedIndexConfig' in result);

// ================================
// ===== READ A CLASS =====
// ================================

// START ReadOneCollection
let classDefinition = await client
  .schema
  .classGetter()
  .withClassName(className)
  .do();

console.log(JSON.stringify(classDefinition, null, 2));
// END ReadOneCollection

// ================================
// ===== READ A CLASS =====
// ================================

// START ReadAllCollections
let allClassDefinitions = await client
  .schema
  .getter()
  .do();

console.log(JSON.stringify(allClassDefinitions, null, 2));
// END ReadAllCollections


// ================================
// ===== UPDATE A CLASS =====
// ================================

// Clean slate
try {
    await client.schema.classDeleter().withClassName(className).do();
  } catch (e) {
    // ignore error if class doesn't exist
  }

// START UpdateCollection
// Define and create a class
const originalClassObj = {
    'class': className,
    'vectorIndexConfig': {
        'distance': 'cosine'  // Note the distance metric
    }
}

// Add the class to the schema
let originalClassResponse = await client
  .schema
  .classCreator()
  .withClass(originalClassObj)
  .do();

const UpdateCollectionObj = {
    'class': className,
    'vectorIndexConfig': {
        'distance': 'dot'  // Note the distance metric
    }
}

// Update the class definition
// Not yet available in TS

File: _includes/code/howto/manage-data.classes.py
# How-to: Manage-Data -> Classes
import os
import json

# ================================
# ===== INSTANTIATION-COMMON =====
# ================================

import weaviate

# Instantiate the client with the OpenAI API key
client = weaviate.connect_to_local(
    port=8080,
    grpc_port=50051,
)

collection_name = "Article"

# ================================
# ===== CREATE A COLLECTION =====
# ================================

# Clean slate
if client.collections.exists(collection_name):
    client.collections.delete(collection_name)

# START CreateCollection
client.collections.create("Article")
# END CreateCollection

# Test
assert client.collections.exists(collection_name)

# ================================
# ===== READ A COLLECTION =====
# ================================

# START ReadOneCollection
articles = client.collections.get("Article")
articles_config = articles.config.get()

print(articles_config)
# END ReadOneCollection

assert articles_config.name == "Article"


# ================================
# ===== READ ALL COLLECTIONS =====
# ================================

# START ReadAllCollections
response = client.collections.list_all()

print(response)
# END ReadAllCollections

assert type(response) == dict
assert collection_name in response


# ================================
# ===== UPDATE A COLLECTION =====
# ================================

# Clean slate
if client.collections.exists(collection_name):
    client.collections.delete(collection_name)

# Define and create a collection
client.collections.create(
    name="Article",
    inverted_index_config=wvc.Configure.inverted_index(
        bm25_k1=1.2
    )
)

# START UpdateCollection
import weaviate.classes as wvc

articles = client.collections.get("Article")
# END UpdateCollection


# Create an object to make sure it remains mutable
for _ in range(5):
    articles.data.insert({
        "title": "A grand day out."
    })
old_config = articles.config.get()
# START UpdateCollection

# Update the collection definition
articles.config.update(
    inverted_index_config=wvc.Reconfigure.inverted_index(
        bm25_k1=1.5
    )
)
# END UpdateCollection

new_config = articles.config.get()

assert old_config.inverted_index_config.bm25.k1 == 1.2
assert new_config.inverted_index_config.bm25.k1 == 1.5
